record/record.utils.js

```diff
@@ -629,14 +629,75 @@
             recordTableUtils.FlowControlObject.call(this, maxRequests);
         }
 
+        function attachGoogleDatasetJsonLd (tuple) {
+            // check the googleDatasetConfigs global variable
+            // that is defined in google-dataset-config.js file
+            var catalogID = tuple.reference.location.catalogId, //getting the catalog-id without version
+                schemaName = tuple.reference.table.schema.name,
+                tableName = tuple.reference.table.name;
+            if (typeof googleDatasetConfigs !== 'undefined' &&
+                DataUtils.isObjectAndNotNull(googleDatasetConfigs) &&
+                DataUtils.isObjectAndNotNull(googleDatasetConfigs[catalogID]) &&
+                DataUtils.isObjectAndNotNull(googleDatasetConfigs[catalogID][schemaName]) &&
+                DataUtils.isObjectAndNotNull(googleDatasetConfigs[catalogID][schemaName][tableName])
+                ) {
+                var currConfig = googleDatasetConfigs[catalogID][schemaName][tableName];
+                if(currConfig) {
+                    var columns = currConfig.columns;
+                    var allValuesArray = currConfig.values;
+                    var valueInTuple;
+                    
+                    if (Array.isArray(columns) && Array.isArray(allValuesArray) && 
+                    (columns.length == allValuesArray.length || (columns.length == 1 && allValuesArray.length > 0))) {
```

Comment:
This condition doesn't make sense to me. We could have 
```
{
  "columns": ["col1", "col2", "col3"],
  "values": [ ["col1_1", "col2_1", "col3_1"]]
}
```
Where `columns.length = 3` and `allValuesArray = 1`.  I don't understand why you're checking for something like that. The following is all you need:

```js
// config defined for the table, but the values are invalid: don't add json-ld
if (!(Array.isArray(columns) || !Array.isArray(allValuesArray)) {
  return;
}

var matchFound = allValuesArray.some(function (v) {
  // make sure the value is an array
  var val = Array.isArray(v) ? v : [v];

  // make sure we have value for all the columns
  if (val.length != columns.length) {
    return false;
  }

  // all the values match
  return columns.every(function (col, index) {
    return tuple.data[col] == val[index];
   });

});

// didn't find a match: don't add json-ld
if (!matchFound) {
  return;
}
```

Reply:
I had misunderstood the config earlier, corrected it now.